library/core/src/slice/mod.rs

```diff
@@ -805,8 +805,9 @@ impl<T> [T] {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     #[inline]
+    #[track_caller]
     pub fn windows(&self, size: usize) -> Windows<'_, T> {
-        let size = NonZeroUsize::new(size).expect("size is zero");
+        let size = NonZeroUsize::new(size).expect("windows cannot have a size of zero");
```

Comment:
The [recommended message style](https://doc.rust-lang.org/std/result/enum.Result.html#recommended-message-style) is to express what the value *should* be, not what it shouldn't.  Therefore:

```suggestion
        let size = NonZeroUsize::new(size).expect("windows should have a non-zero size");
```